iceoryx_binding_c/include/iceoryx_binding_c/user_trigger.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 by Robert Bosch GmbH, Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_BINDING_C_USER_TRIGGER_H
+#define IOX_BINDING_C_USER_TRIGGER_H
+
+#include "iceoryx_binding_c/enums.h"
+#include "iceoryx_binding_c/internal/c2cpp_binding.h"
+#include "iceoryx_binding_c/types.h"
+
+/// @brief user trigger handle
+typedef CLASS UserTrigger* iox_user_trigger_t;
+typedef CLASS WaitSet* iox_ws_t;
+
+/// @brief initialize user trigger handle
+/// @param[in] self pointer to preallocated memory of size = sizeof(iox_user_trigger_storage_t)
+/// @return handle to user trigger
+iox_user_trigger_t iox_user_trigger_init(iox_user_trigger_storage_t* self);
```

Comment:
Didn't we want to use create and destroy?

Reply:
I slightly remember the discussion. And the main point was `create` allocates memory for you and initializes the underlying class with a placement new and `destroy` releases this memory.

Our current approach is that we are providing structs for the user so that there is no problem to acquire memory for those classes on the stack the only thing left to do for the user is to initialize that memory with a placement new. And this is what the `init` function does. There is no memory allocated only an placement new is performed on a user provided piece of memory.

Reply:
Yes, create and destroy would be the names if you allocate and free the memory. Then it's fine